specification/common-types/resource-management/v5/networksecurityperimeter.json

```diff
@@ -0,0 +1,400 @@
+{
+  "swagger": "2.0",
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "info": {
+    "version": "2023-09-15-preview",
+    "title": "Network security perimeter common type definitions",
+    "description": "Common types for network security perimeters based on a shared API specification. These common, versioned type definitions are intended for resource providers (RPs, except Network RP) to use, and reuse, for defining their own API versions that share a set of type definitions that is consistent across providers.",
+    "contact": {}
+  },
+  "paths": {},
+  "definitions": {
+    "PublicNetworkAccess": {
+      "type": "string",
+      "description": "Allow, disallow, or let network security perimeter configuration control public network access to the protected resource. Value is optional but if passed in, it must be 'Enabled', 'Disabled' or 'SecuredByPerimeter'.",
+      "enum": [
+        "Enabled",
+        "Disabled",
+        "SecuredByPerimeter"
+      ],
+      "x-ms-enum": {
+        "name": "PublicNetworkAccess",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "Enabled",
+            "description": "Allows public network access to the resource"
+          },
+          {
+            "value": "Disabled",
+            "description": "Disallows public network access to the resource"
+          },
+          {
+            "value": "SecuredByPerimeter",
+            "description": "The network security perimeter configuration rules allow or disallow public network access to the resource. Requires an associated network security perimeter."
+          }
+        ]
+      }
+    },
+    "NetworkSecurityPerimeterConfigurationListResult": {
+      "description": "Result of a list NSP (network security perimeter) configurations request.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Array of network security perimeter results.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NetworkSecurityPerimeterConfiguration"
+          }
+        },
+        "nextLink": {
+          "description": "The link used to get the next page of results.",
+          "type": "string",
+          "format": "uri"
+        }
+      }
+    },
+    "NetworkSecurityPerimeterConfiguration": {
```

Comment:
It seems like the 'Configuration' here unnecessary, and it is not part of the API, but makes from some really long schema names (and model names for clients), can we remove 'Configuration' from the schema names?

Reply:
It aligns with the intended resource type name in the path, although that is not part of this common spec.

Reply:
If NetworkSecurityPerimiter and NetworkSecurityPermimeterConfiguration are disctinct resources that exist, then this name is absolutely necessary.  Otherwise, it seems unnecessarily long.

Reply:
Actually, we can tell from the definitions that these are two separate things, so I think this name is unfortunate, but necessary.